cmd/sqlflowserver/main_test.go

```diff
@@ -2000,6 +2002,42 @@ FROM %s LIMIT 5;
 	a.NoError(checkWorkflow(ctx, cli, stream))
 }
 
+func CaseWorkflowTrainAndPredictDNNCustomImage(t *testing.T) {
+	if os.Getenv("SQLFLOW_submitter") != "pai" && os.Getenv("SQLFLOW_submitter") != "alisa" {
+		t.Skip("Skip PAI case.")
+	}
+	a := assert.New(t)
+	// use the default image to test
+	customImage := os.Getenv("SQLFLOW_WORKFLOW_STEP_IMAGE")
+	sqlProgram := fmt.Sprintf(`
+SELECT * FROM %s LIMIT 10;
+
+SELECT * FROM %s
+TO TRAIN %s/DNNClassifier
+WITH
+	model.n_classes = 3,
+	model.hidden_units = [64, 32],
+	validation.select = "SELECT * FROM %s"
+LABEL class
+INTO test_workflow_model;`, caseTrainTable, caseTrainTable, customImage, caseTestTable)
```

Comment:
Do we need to add tests on PREDICT/EXPLAIN?

Reply:
Unfortunately predict and explain are not supported until we can find a way to analyze the SQL program and find out the image used by train and apply it to the predict SQL statement.